### PR TITLE
Add React 15.4.0 compatibility

### LINKED ISSUE #111: Incompatibility with React 15.4.0
Your dependency on `react-addons-perf` is breaking with React 15.4.0.

Can this be updated?

### PR DESCRIPTION
**Pull request by joemckie**
Since React 15.4.0, `react-addons-perf` looks to `react-dom`. This PR updates dependency versions and removes the `react-addons-linked-state` dependency, which is now deprecated.

Fixes #111.

### CODE REVIEW

**Comment by RealOrangeOne** [comment]
Thanks for doing this (and catching the linked-state deprication!). Not sure what's up with the tests failing. Am having a look into that now

**Comment by joemckie** [comment]
@RealOrangeOne no problem! I'm not sure what's up with the tests either. Looks like it's only on v4, but looks like a problem that's been going on for a while (look at the previous build logs).

File: package.json [hunk 1]
```diff
@@ -42,22 +42,22 @@
     "eslint-plugin-react": "=5.1.1",
     "eslint-plugin-react-native": "=1.0.2",
     "mocha": "=2.5.3",
-    "react": "=15.1.0",
+    "react": "=15.4.0",
     "react-native": "^0.27.0"
```
Comment (RealOrangeOne) [inline package.json]: looks like tests are breaking here. Please change to >=0.27.0. Not sure why this was only happening in node 4 though

Comment (joemckie) [reply package.json]: Updated 👍

File: package.json [hunk 2]
```diff
@@ -42,22 +42,22 @@
...
+    "cubic-bezier": "^0.1.2",
+    "invariant": "^2.2.1",
+    "keymirror": "^0.1.1",
+    "raf": "^3.2.0",
+    "react-addons-create-fragment": "^15.4.0",
+    "react-addons-perf": "^15.4.0",
+    "react-addons-pure-render-mixin": "^15.4.0",
+    "react-addons-test-utils": "^15.4.0",
+    "react-addons-update": "^15.4.0",
+    "react-dom": "^15.4.0",
```
Comment (RealOrangeOne) [inline package.json:58]: do we need to explicitly state a dependancy on react-dom? Surely the addons libraries will handle installing this for us?

Comment (joemckie) [reply package.json:58]: The addons specify a peerDependency so they don't install `react-dom` unfortunately!

**Comment by RealOrangeOne** [comment]
seems the reason the tests are failing is because react-native doesnt yet support react@15.4.0, so merging this would likely break the install for a lot of people

**Comment by joemckie** [comment]
@RealOrangeOne looks like you might be right.

[v0.38.0-rc.0](https://github.com/facebook/react-native/releases/tag/v0.38.0-rc.0) seems to have support for v15.4.0, and RN has a fortnightly release cycle, so I'd expect compatibility to be available in a few days' time.

**Review by RealOrangeOne — APPROVED** [review]
Pending next stable release of react-native

**Comment by joemckie** [comment]
@RealOrangeOne : [v0.38.0](https://github.com/facebook/react-native/releases/tag/v0.38.0)

File: package.json [hunk 3]
```diff
@@ -28,22 +28,22 @@
...
-    "mocha": "=2.5.3",
+    "eslint-config-airbnb": "9.0.1",
+    "eslint-plugin-import": "1.8.0",
+    "eslint-plugin-jsx-a11y": "1.2.2",
+    "eslint-plugin-react": "5.1.1",
+    "eslint-plugin-react-native": "1.0.2",
+    "mocha": "2.5.3",
     "react": "^15.4.0",
-    "react-native": ">=0.27.0"
+    "react-native": "0.38.0"
```
Comment (RealOrangeOne) [inline package.json]: Can you leave this as a loose dependancy, so it upgrades automatically. Will help with tests

File: package.json [hunk 4]
```diff
@@ -28,22 +28,22 @@
   },
   "homepage": "https://github.com/RealOrangeOne/react-native-mock#readme",
   "devDependencies": {
-    "babel": "=6.5.2",
```
Comment (RealOrangeOne) [inline package.json:31]: Is there a reason you removed the _=_

Comment (joemckie) [reply package.json:31]: Had a read through the docs and it isn't necessary to include the `=`, this is done automatically by NPM

https://github.com/npm/node-semver#versions

> A leading "=" or "v" character is stripped off and ignored.

**Comment by joemckie** [comment]
Hey FYI @RealOrangeOne the NPM package still needs version bumping to get this fix out :)

**Comment by messense** [comment]
Please release a new version on NPM. Thanks @RealOrangeOne

**Comment by joemckie** [comment]
Yup. Currently blocking my project and I'm having to workaround right now!

**Comment by RealOrangeOne** [comment]
Released v0.2.9

**Pull request merged**